src/components/Video/VideoContent/VideoContent.tsx

```diff
@@ -0,0 +1,190 @@
+import React, { useState, useEffect, useRef, useCallback } from 'react';
+import videojs from 'video.js';
+import Component from 'video.js/dist/types/component.d';
+import Player from 'video.js/dist/types/player.d';
+import classNames from 'classnames';
+import { useSearchParams } from 'next/navigation';
+import styles from './VideoContent.module.scss';
+import LazyLoad from '../../LazyLoad/LazyLoad';
+import useSharedUnmutedVideoState from '../useSharedUnmutedVideoState';
+import 'video.js/dist/video-js.css';
+import Controls from '../Controls/Controls';
+import { VideoData } from '../../../types/types';
+
+export type VideoContentProps = {
+  hasControls: boolean;
+  isAutoplaying: boolean;
+  isLooping: boolean;
+  video: VideoData;
+};
+
+type VideoJsComponent = Component & {
+  handleClick: () => void;
+};
+
+const VideoContent = ({
+  isAutoplaying,
+  hasControls,
+  isLooping,
+  video: { baseUrl, blur, height, hls, width },
+}: VideoContentProps) => {
+  const videoContainerRef = useRef(null);
+
+  const searchParams = useSearchParams();
+  const hasFocusMode = searchParams.get('hasFocusMode');
+
+  const [player, setPlayer] = useState<Player>(null);
+  const [isPlaying, setIsPlaying] = useState(isAutoplaying);
+  const [hasLoaded, setHasLoaded] = useState(false);
+
+  const [isMuted, toggleIsMuted] = useSharedUnmutedVideoState(hls ?? 'unknown');
+
+  const handlePlayToggle = () => {
+    if (!player) {
+      return;
+    }
+
+    if (player.paused()) {
+      player.play();
+
+      return;
+    }
+
+    player.pause();
+  };
+
+  const handleLoadVideo = useCallback(() => {
+    // Have to do it again manually here, since all searchParams from the
+    // useSearchParams hook are undefined on initial render
+    const searchParamsObject = new URLSearchParams(window.location.search);
+    const newControls = searchParamsObject.get('hasFocusMode');
+
+    const videoElement = document.createElement('video-js');
+    videoContainerRef.current.appendChild(videoElement);
+
+    const videoJsPlayer = videojs(videoElement, {
```

Comment:
Don't we need to destroy these videojs instances? E.g. when the component unmounts do they just get garbage collected because their videoElement is gone?